### PR TITLE
Using redux middleware connection and subscribing from server

### PR DESCRIPTION
**Pull request by revathskumar**
- Moving the zmq connection and subscribing logic to redux middleware
- Added functionality to start CPU profiling
- Some styleguide fixes

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["es2015", "react"]
+  "presets": ["es2015", "react", "stage-0"]
```
Comment (emilsoman) [inline .babelrc:2]: What's stage-0 added for ?

File: app/components/layout.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-import Navbar from '../components/common/Navbar'
-import Row from './row'
-import {connectToServer, handshake} from '../redux/actions'
-import DisplayHandshake from './displayhandshake'
-import Toolbelt from './toolbelt'
+import React, {PropTypes} from 'react';
+import {connect} from 'react-redux';
+// import {bindActionCreators} from 'redux';
+import Navbar from '../components/common/Navbar';
+// import Row from './row';
```
Comment (emilsoman) [inline app/components/layout.js:5]: Can we remove commented lines above ?

File: app/components/layout.js [hunk 3]
```diff
@@ -19,14 +19,18 @@ class Layout extends React.Component {
...
+  cpuProfiling() {
+    this.props.dispatch(startCpuProfiling());
   }
 
   render() {
-    return(
+    return (
       <div>
-        <Navbar {...this.props} connect={this.connect}/>
+        <Navbar {...this.props} connect={::this.connect}/>
```
Comment (emilsoman) [inline app/components/layout.js:33]: Can you explain this change ? Not all of us are familiar with `::`

File: .babelrc (hunk 1 above)
Comment (revathskumar) [inline .babelrc:2]: For Destructuring assignment.

Comment (kgrz) [inline .babelrc:2]: Destructuring is provided by es2015 isn't it?  stage-0 provides `bind` syntax and a `do` statement.

File: app/redux/reducers/connection.js [hunk 4]
```diff
@@ -19,10 +19,27 @@ const connection = function (state = initialState, action) {
...
+      return action.data;
+    default:
+      return state;
+  }
+};
+
+const cpuProfile = (state = {status: 'STOPPED', data: []}, action) => {
+  switch (action.type) {
+    case 'DATA':
+      return Object.assign({}, state, {status: 'RUNNING', data: [...state.data, action.data], from: action.from});
```
Comment (emilsoman) [inline app/redux/reducers/connection.js:33]: What's `from` here ?

File: app/components/layout.js (hunk 2 above)
Comment (revathskumar) [inline app/components/layout.js:5]: I removed this in [another commit](https://github.com/code-mancers/rbkit-electron-redux/pull/5/files#diff-ebda658fb74330dd9e6b1cb58b9ad64fR6) which is in #5

File: app/redux/store.js [hunk 5]
```diff
@@ -1,16 +1,86 @@
...
-let createStoreWithMiddleware;
+const requester = ZMQ.socket('req');
+const subscriber = ZMQ.socket('sub');
 
-createStoreWithMiddleware = compose(
-  applyMiddleware(logger(), thunk)
-)(createStore)
+const zmqMiddlewareCreator = (zmq, subzmq) => {
+  return ({dispatch}) => {
+    return next => action => {
```
Comment (emilsoman) [inline app/redux/store.js:13]: What's `next => action => {}` for ?

File: app/components/layout.js (hunk 3 above)
Comment (siddharthbhagwan) [inline app/components/layout.js:33]: This is es7 syntax for `this.connect().bind(this)` afaik

File: .babelrc (hunk 1 above)
Comment (revathskumar) [inline .babelrc:2]: Sorry that's for Spread operator used [here](https://github.com/code-mancers/rbkit-electron-redux/pull/4/files#diff-58ed19aaca2d7ef6489531b4b3594890R21).

File: app/redux/store.js (hunk 5 above)
Comment (siddharthbhagwan) [inline app/redux/store.js:13]: Its shorthand for

```
return (next) => {
  return (action) => {
     .
     .
     .
  }
}
```

used when middleware is dispatching an action

File: app/components/layout.js (hunk 3 above)
Comment (revathskumar) [inline app/components/layout.js:33]: yes similar to `this.connect.bind(this)`. But as we discussed in chatroom we won't be using this any more. I also remove this is some other commit.

File: app/redux/store.js (hunk 5 above)
Comment (emilsoman) [inline app/redux/store.js:13]: Yes, but why do we do this ? (trying to understand how this gets consumed)

File: app/redux/reducers/connection.js (hunk 4 above)
Comment (revathskumar) [inline app/redux/reducers/connection.js:33]: Data from requester or subscriber.

File: app/redux/store.js (hunk 5 above)
Comment (revathskumar) [inline app/redux/store.js:13]: AFAIK every middleware is curried function. and `applyMiddleware` accepts a series of such functions and maps through it.

I think a part of magic happens [here](https://github.com/reactjs/redux/blob/master/src/applyMiddleware.js#L29-L30).

**Pull request merged**